agent/rpc/peering/service.go

```diff
@@ -399,6 +402,21 @@ func (s *Server) Establish(
 	return resp, nil
 }
 
+// validatePeeringInPartition makes sure that we don't create a peering in the same partition. We do so by checking
+// to see if the current server's server addresses slice has any intersection with the slice from the token.
+func (s *Server) validatePeeringInPartition(remotePeerID, partition string) error {
+	_, peering, err := s.Backend.Store().PeeringReadByID(nil, remotePeerID)
+	if err != nil {
+		return fmt.Errorf("cannot validate peering in partition: %w", err)
```

Comment:
should just return err instead of wrapping (misleading error message)

Reply:
I think it's worth saying the "why" the `PeeringReadByID` is made in the first place.

As a user here, the UX of the "why" (eg `cannot validate peering in partition: error_msg`) let's me know that there is a validation issue with my  request.

Reply:
In that case, a general rule of thumb I like is:
Wrapping error messages should contextualize where the error came from (and avoid self-referencing their containing functions).

In this case, the failure is `PeeringReadByID`, so it's misleading to prematurely wrap the error.
```suggestion
		return fmt.Errorf("cannot read peering by ID: %w", err)
```

The actual context should be wrapped by the caller
```go
	if err := s.validatePeeringInPartition(tok.PeerID, req.Partition); err != nil {
		return nil, fmt.Errorf("cannot validate peering in partition: %w", err)
	}
```